server/jetty/src/main/java/io/deephaven/server/jetty/ControlledCacheResource.java

```diff
@@ -51,11 +48,16 @@ public boolean isDirectory() {
     }
 
     @Override
-    public long lastModified() {
-        // Always return -1, so that we don't get the build system timestamp. In theory we could return the app startup
+    public boolean isReadable() {
+        return wrapped.isReadable();
+    }
+
+    @Override
+    public Instant lastModified() {
+        // Always return -1, so that we don't get the build system timestamp. In theory, we could return the app startup
```

Comment:
It looks like `Instant.ofEpochMilli(-1)` is still treated as a _special_ (as was -1 in Jetty 11), even though it's not documented in the javadoc, at least as used by `org.eclipse.jetty.server.ResourceService`:

```java
                long ifmsl = HttpDateTime.parseToEpoch(ifms);
                if (ifmsl != -1)
                {
                    long lm = content.getResource().lastModified().toEpochMilli();
                    if (lm != -1 && lm / 1000 <= ifmsl / 1000)
                    {
                        writeHttpError(request, response, callback, HttpStatus.NOT_MODIFIED_304);
                        return true;
                    }
                }
```

I'm not sure if we actually depend on this behavior, or we just want an old timestamp.

Reply:
We don't explicitly depend on _this_ behavior, but we want to ensure that we control the caching through something other than "how old is the file". The last-modified value is stored by the client, and then the server (as you quoted) will send the file contents if the file has changed since that date. 

Picking a constant is no good then, since that constant will always be the same, the server will never send a new version. Reading the timestamp from the file on disk is an option (and is the default behavior), but this bites us in another way: a release series might look like
 * 1/1/2024 release 3.1
 * 1/15/2024 release 4.0
 * 2/1/2024 release 3.2

If you update from 3.1 to 3.2 and then to 4.0, the files get _older_, so we fail this check.

As such, -1 to skip sending the last modified at all is probably the best bet. Other options we considered at the time: record the server start time, or server deployment time (somehow), but -1 still seems safer.